filebeat/crawler/prospector.go

```diff
@@ -35,9 +52,48 @@ func (p *Prospector) Init() error {
 		return err
 	}
 
+	var prospectorer Prospectorer
+
+	switch p.ProspectorConfig.Harvester.InputType {
+	case cfg.StdinInputType:
+		prospectorer, err = NewProspectorStdin(p.ProspectorConfig, p.channel)
+		prospectorer.Init()
+	case cfg.LogInputType:
+		prospectorer, err = NewProspectorLog(p.ProspectorConfig, p.channel, p.registrar)
+		prospectorer.Init()
+	default:
+		return fmt.Errorf("Invalid prospector type: %v", p.ProspectorConfig.Harvester.InputType)
+	}
+
+	p.prospectorer = prospectorer
+
 	return nil
 }
 
+// Starts scanning through all the file paths and fetch the related files. Start a harvester for each file
+func (p *Prospector) Run() {
+
+	// This signals we finished considering the previous state
+	event := &input.FileState{
+		Source: nil,
+	}
+	p.registrar.Persist <- event
```

Comment:
I'm confused. Isn't this nil-even used by log-prospector to signal back it finished the initial scan on files alread in registrar?